File: server/src/main/java/mojac/musicnode/domain/Playlist.java
package mojac.musicnode.domain;

import jakarta.persistence.*;
import lombok.Getter;
import mojac.musicnode.domain.thumb.Thumbs;

import java.util.List;

@Entity
@Getter
public class Playlist {
    @Id @GeneratedValue
    @Column(name = "music_list_id")
    private Long id;

    private String contents; // JSON String

    private String name;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id")
    private Member member;

    @OneToMany(mappedBy = "playlist")
    private List<Thumbs> thumbs;

    protected Playlist() {}

    public Playlist(String name, String contents, Member member) {
        this.name = name;
        this.contents = contents;
        this.member = member;
    }

    public void rename(String name) {
        this.name = name;
    }
}

File: client/src/domain/Playlist.ts
import { MusicInfo } from './MusicInfo';

interface PlaylistConstructorProps {
  id: number;
  name: string;
  contents: string;
}

export class Playlist {
  id?: number;
  name?: string;
  contents: MusicInfo[];

  constructor({ id, name, contents }: PlaylistConstructorProps) {
    this.id = id;
    this.name = name;
    this.contents = Object.values(JSON.parse(contents));
  }
}
